fastestimator/pipeline.py

```diff
@@ -139,102 +139,111 @@ def get_per_epoch(ops: Iterable[Union[T, Scheduler[T]]], epoch: int) -> List[T]:
 
 
 class BasePipeline:
-    def get_modes(self) -> Set[str]:
-        raise NotImplementedError()
-
-    def get_global_batch_size(self, mode: str, epoch: int) -> int:
-        raise NotImplementedError()
-
-    def get_signature_epochs(self, mode: str) -> List[int]:
-        raise NotImplementedError()
-
-    def get_all_output_keys(self) -> Set[str]:
```

Comment:
what is your plan for later validating whether traces will plausibly work given their input requirements?